id3: fix reading/writing v1.0 and v1.1

corubba opened the pull request:
Properly read and write ID3 v1.0 and v1.1 tags according to the [spec](https://id3.org/ID3v1). ID3 v1.0 tags are no longer always read as v1.1, which previously resulted in wrong track numbers. Writing will use v1.0 by default, automatically upgrading to v1.1 if needed.

I am not entirely sure what the WinAmp special-case was for, but the (also fixed) testcase still passes so ... probably still covered.

Fixes #668

phw reviewed (approved):
Good fix, thank you very much. LGTM

phw committed:
id3: fix reading/writing v1.0 and v1.1

Properly read and write ID3 v1.0 and v1.1 tags according to the spec
[0]. ID3v1.0 tags are no longer always read as v1.1, which previously
resulted in wrong track numbers. Writing will use v1.0 by default,
automatically upgrading to v1.1 if needed.

I am not entirely sure what the WinAmp special-case was for, but the
(also fixed) testcase still passes so ... probably still covered.

[0] https://id3.org/ID3v1

phw commented:
I took the freedom of rebasing this PR to resolve the conflict with the changes in https://github.com/quodlibet/mutagen/pull/709
